### PR TITLE
Revise dpu operator jobs

### PR DESCRIPTION
**Pull request by wizhaoredhat**
The e2e-aws jobs takes 1.5 Hours to complete and is not useful for this project yet; thus was removed. Added job to build images and run make test. The later has yet to be implemented.

### CODE REVIEW

**Comment by wizhaoredhat** [comment]
@dhellmann PTAL

File: ci-operator/config/openshift/dpu-operator/openshift-dpu-operator-main.yaml [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+images:
+- dockerfile_path: Dockerfile.rhel
+  inputs:
+    ocp_4.16_base-rhel9:
+      as:
+      - registry.ci.openshift.org/ocp/4.16:base-rhel9
+    ocp_builder_rhel-9-golang-1.21-openshift-4.16:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16
+  to: dpu-operator
```
Comment (dhellmann) [inline ci-operator/config/openshift/dpu-operator/openshift-dpu-operator-main.yaml:21]: I'm still a bit worried that this name is going to be confusingly similar to `dpu-network-operator` but I don't have any better suggestions for you.

**Review by dhellmann — COMMENTED** [review]
The promoted image name is acceptable. I'll leave it up to your team to approve any of the other changes to CI.

/approve

File: ci-operator/config/openshift/dpu-operator/openshift-dpu-operator-main.yaml (hunk 1 above)
Comment (wizhaoredhat) [reply ci-operator/config/openshift/dpu-operator/openshift-dpu-operator-main.yaml:21]: Thanks Doug, I think hopefully at somepoint we can fully deprecate `dpu-network-operator`; thus the confusion will only be temporary.

File: ci-operator/config/openshift/dpu-operator/openshift-dpu-operator-main.yaml [hunk 2]
```diff
@@ -30,10 +41,10 @@ resources:
       cpu: 100m
       memory: 200Mi
 tests:
-- as: e2e-aws
-  steps:
-    cluster_profile: aws
-    workflow: openshift-e2e-aws
+- as: unit
```
Comment (wizhaoredhat) [inline ci-operator/config/openshift/dpu-operator/openshift-dpu-operator-main.yaml:44]: @bn222

**Pull request merged**